backend/src/main/java/org/eclipse/tractusx/puris/backend/common/ddtr/logic/util/DtrRequestBodyBuilder.java

```diff
@@ -0,0 +1,252 @@
+        body.set("specificAssetIds", specificAssetIdsArray);
+
+        var partnerRefNode = List.of(getReferenceObject(materialPartnerRelation.getPartner().getBpnl()));
+
+        var digitalTwinObject = getDigitalTwinObject(partnerRefNode);
+        specificAssetIdsArray.add(digitalTwinObject);
+
+        var manufacturerIdObject = getManufacturerIdObject(materialPartnerRelation.getPartner().getBpnl(), partnerRefNode);
+        specificAssetIdsArray.add(manufacturerIdObject);
+
+        var manufacturerPartIdObject = getManufacturerPartIdObject(materialPartnerRelation.getPartnerMaterialNumber(), partnerRefNode);
+        specificAssetIdsArray.add(manufacturerPartIdObject);
+
+        var customerPartIdObject = getCustomerPartIdObject(material.getOwnMaterialNumber(), materialPartnerRelation.getPartner().getBpnl());
+        specificAssetIdsArray.add(customerPartIdObject);
+
+        var submodelDescriptorsArray = objectMapper.createArrayNode();
+        body.set("submodelDescriptors", submodelDescriptorsArray);
+
+        var itemStockRequestSubmodelObject = getItemStockSubmodelObject();
+        submodelDescriptorsArray.add(itemStockRequestSubmodelObject);
+        log.info("Created body for material " + material.getOwnMaterialNumber() + "\n" + body.toPrettyString());
+        return body;
+    }
+
+    /**
+     * Creates a RequestBody for registering or updating a <b>Product-Type</b> Material
+     *
+     * @param material          The Material
+     * @param productTwinId     The ProductTwinId
+     * @param mprs              The list of all MaterialProductRelations that exist with customers of the given Material
+     * @return                  The Request Body
+     */
+    public JsonNode createProductRegistrationRequestBody(Material material, String productTwinId, List<MaterialPartnerRelation> mprs) {
+        var body = objectMapper.createObjectNode();
+        body.put("id", productTwinId);
+        body.put("globalAssetId", material.getMaterialNumberCx());
+        var specificAssetIdsArray = objectMapper.createArrayNode();
+        body.set("specificAssetIds", specificAssetIdsArray);
+        mprs = mprs.stream().filter(MaterialPartnerRelation::isPartnerBuysMaterial).filter(mpr -> mpr.getMaterial().equals(material)).toList();
+        var partnerRefObjects = mprs.stream().map(mpr -> getReferenceObject(mpr.getPartner().getBpnl())).toList();
+        if (material.isProductFlag()) {
+            var digitalTwinObject = getDigitalTwinObject(partnerRefObjects);
+            specificAssetIdsArray.add(digitalTwinObject);
+            var manufacturerPartIdObject = getManufacturerPartIdObject(material.getOwnMaterialNumber(), partnerRefObjects);
+            specificAssetIdsArray.add(manufacturerPartIdObject);
+            var manufacturerIdObject = getManufacturerIdObject(variablesService.getOwnBpnl(), partnerRefObjects);
+            specificAssetIdsArray.add(manufacturerIdObject);
+            for (var mpr : mprs) {
```

Comment:
I added the following code to the supplier role setup in DataInjectionCommandLineRunner to simulate 2 customers for the supplier for the same material in line 263
``` java
        Partner nonScenarioPartner = createAndGetNonScenarioCustomer();
        MaterialPartnerRelation semiconductorPartnerRelation2 = new MaterialPartnerRelation(semiconductorMaterial,
            nonScenarioPartner, "FOO_BAR", false, true);
        semiconductorPartnerRelation2.setPartnerCXNumber(semiconductorMatNbrCatenaX);
        semiconductorPartnerRelation2 = mprService.create(semiconductorPartnerRelation);

        log.info("Created Relation " + semiconductorPartnerRelation2);
```

From my understanding the CX ID should be the same (supplier CX ID is leading) and customer part id should be different in the mprs. When rebuilding and spinning up the application only one customerPartId node will be found in the shellDescriptor and not two.

Please check

Reply:
This line in your code sample is not correct: 
```
semiconductorPartnerRelation2 = mprService.create(semiconductorPartnerRelation);
```

This means that the new MaterialPartnerRelation was not created at the MPR Service (instead the other MPR was created twice) and therefore no registration at the dDTR was triggered. 

It should be 
```
semiconductorPartnerRelation2 = mprService.create(semiconductorPartnerRelation2);
```
instead. 

After this change, there are two customerPartId`s for that product, as expected.